subsys/bluetooth/host/classic/avrcp.c

```diff
@@ -3720,46 +3917,435 @@ int bt_avrcp_tg_send_subunit_info_rsp(struct bt_avrcp_tg *tg, uint8_t tid)
+{
+	struct net_buf *buf;
+	int err;
+
+	if ((tg == NULL) || (tg->avrcp == NULL)) {
+		LOG_ERR("Invalid AVRCP target");
+		return -EINVAL;
+	}
+
+	if (!IS_TG_ROLE_SUPPORTED()) {
+		LOG_ERR("Target role not supported");
+		return -ENOTSUP;
+	}
+
+	buf = avrcp_create_browsing_pdu(tg->avrcp);
+	if (buf == NULL) {
+		return -ENOBUFS;
+	}
+
+	if (net_buf_tailroom(buf) < sizeof(status) +
+	   ((status == BT_AVRCP_STATUS_OPERATION_COMPLETED) ? sizeof(num_items) : 0U)) {
+		LOG_ERR("Not enough tailroom for CHANGE_PATH rsp");
+		net_buf_unref(buf);
+		return -ENOMEM;
+	}
+
+	net_buf_add_u8(buf, status);
+	if (status == BT_AVRCP_STATUS_OPERATION_COMPLETED) {
+		net_buf_add_be32(buf, num_items);
+	}
+
+	err = bt_avrcp_browsing_send(tg->avrcp, buf, BT_AVRCP_PDU_ID_CHANGE_PATH,
+				     BT_AVCTP_RESPONSE, tid);
+	if (err < 0) {
+		LOG_ERR("Failed to send CHANGE_PATH rsp (err: %d)", err);
+		net_buf_unref(buf);
+	}
+	return err;
+}
+
+int bt_avrcp_tg_get_item_attrs(struct bt_avrcp_tg *tg, uint8_t tid, struct net_buf *buf)
+{
+	int err;
+
+	if ((tg == NULL) || (tg->avrcp == NULL) || (buf == NULL)) {
+		LOG_ERR("Invalid AVRCP target");
+		return -EINVAL;
+	}
+
+	if (!IS_TG_ROLE_SUPPORTED()) {
+		LOG_ERR("Target role not supported");
+		return -ENOTSUP;
+	}
+
+	err = bt_avrcp_browsing_send(tg->avrcp, buf, BT_AVRCP_PDU_ID_GET_ITEM_ATTRS,
+				     BT_AVCTP_RESPONSE, tid);
+	if (err < 0) {
+		LOG_ERR("Failed to send GET_ITEM_ATTRIBUTES rsp (err: %d)", err);
+	}
+	return err;
+}
+
+int bt_avrcp_tg_get_total_number_of_items(struct bt_avrcp_tg *tg, uint8_t tid,
+						   struct net_buf *buf)
+{
+	int err;
+
+	if ((tg == NULL) || (tg->avrcp == NULL) || (buf == NULL)) {
+		LOG_ERR("Invalid AVRCP target");
+		return -EINVAL;
+	}
+
+	if (!IS_TG_ROLE_SUPPORTED()) {
+		LOG_ERR("Target role not supported");
+		return -ENOTSUP;
+	}
+
+	err = bt_avrcp_browsing_send(tg->avrcp, buf, BT_AVRCP_PDU_ID_GET_TOTAL_NUMBER_OF_ITEMS,
+				     BT_AVCTP_RESPONSE, tid);
+	if (err < 0) {
+		LOG_ERR("Failed to send GET_TOTAL_NUMBER_OF_ITEMS rsp (err: %d)", err);
+	}
+	return err;
+}
+
+int bt_avrcp_tg_search(struct bt_avrcp_tg *tg, uint8_t tid, struct net_buf *buf)
+{
+	int err;
+
+	if ((tg == NULL) || (tg->avrcp == NULL) || (buf == NULL)) {
+		LOG_ERR("Invalid AVRCP target");
+		return -EINVAL;
+	}
+
+	if (!IS_TG_ROLE_SUPPORTED()) {
+		LOG_ERR("Target role not supported");
+		return -ENOTSUP;
+	}
+
+	err = bt_avrcp_browsing_send(tg->avrcp, buf, BT_AVRCP_PDU_ID_SEARCH,
+				     BT_AVCTP_RESPONSE, tid);
+	if (err < 0) {
+		LOG_ERR("Failed to send SEARCH rsp (err: %d)", err);
+	}
+	return err;
+}
+
+int bt_avrcp_tg_browsing_general_reject(struct bt_avrcp_tg *tg, uint8_t tid, uint8_t status)
+{
+	struct net_buf *buf;
+	int err;
+
+	if ((tg == NULL) || (tg->avrcp == NULL)) {
+		LOG_ERR("Invalid AVRCP target");
+		return -EINVAL;
+	}
+	if (!IS_TG_ROLE_SUPPORTED()) {
+		LOG_ERR("Target role not supported");
+		return -ENOTSUP;
+	}
```

Comment:
I see that the similar codes exist for many functions. Suggest to use small function to do it, then make the codes more simple.